staging/src/k8s.io/kube-apiextensions-server/test/integration/registration_test.go

```diff
@@ -0,0 +1,188 @@
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+	apiextensionsv1alpha1 "k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/v1alpha1"
+	"k8s.io/kube-apiextensions-server/test/integration/testserver"
+)
+
+func instantiateNoxuResource(t *testing.T, namespace, name string, client *dynamic.ResourceClient, definition *apiextensionsv1alpha1.CustomResource) *unstructured.Unstructured {
+	noxuInstanceToCreate := testserver.NewNoxuInstance(namespace, name)
+	createdNoxuInstance, err := client.Create(noxuInstanceToCreate)
+	if err != nil {
+		t.Logf("%#v", createdNoxuInstance)
+		t.Fatal(err)
+	}
+	createdObjectMeta, err := meta.Accessor(createdNoxuInstance)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// it should have a UUID
+	if len(createdObjectMeta.GetUID()) == 0 {
+		t.Errorf("missing uuid: %#v", createdNoxuInstance)
+	}
+	createdTypeMeta, err := meta.TypeAccessor(createdNoxuInstance)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if e, a := definition.Spec.Group+"/"+definition.Spec.Version, createdTypeMeta.GetAPIVersion(); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+	if e, a := definition.Spec.Names.Kind, createdTypeMeta.GetKind(); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+	return createdNoxuInstance
+}
+
+func TestMultipleRegistration(t *testing.T) {
+	stopCh, apiExtensionClient, clientPool, err := testserver.StartDefaultServer()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer close(stopCh)
+
+	ns := "not-the-default"
+	noxuDefinition := testserver.NewNoxuCustomResourceDefinition()
+	noxuVersionClient, err := testserver.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, clientPool)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	noxuNamespacedResourceClient := noxuVersionClient.Resource(&metav1.APIResource{
+		Name:       noxuDefinition.Spec.Names.Plural,
+		Namespaced: noxuDefinition.Spec.Scope == apiextensionsv1alpha1.NamespaceScoped,
+	}, ns)
+	noxuList, err := noxuNamespacedResourceClient.List(metav1.ListOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	noxuListListMeta, err := meta.ListAccessor(noxuList)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	noxuNamespacedWatch, err := noxuNamespacedResourceClient.Watch(metav1.ListOptions{ResourceVersion: noxuListListMeta.GetResourceVersion()})
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer noxuNamespacedWatch.Stop()
+
+	instances := map[string]*struct {
+		Added    bool
+		Deleted  bool
+		Instance *unstructured.Unstructured
+	}{
+		"foo": {},
+		"bar": {},
+	}
+
+	for key, val := range instances {
+		val.Instance = instantiateNoxuResource(t, ns, key, noxuNamespacedResourceClient, noxuDefinition)
+	}
+
+	addEvents := 0
+	for addEvents < len(instances) {
```

Comment:
actually, I think all the errors in here are fatal.  The rest of the test doesn't function properly if this part fails.